Admin & User Settings UI Updates

mwodahl opened the pull request:
# PR Details

This PR includes styling updates to the User Settings and Admin pages to reflect the following Figma prototype: https://www.figma.com/proto/QG64ugNXpXMUk2vQ1wgPdZ/CV-Manager-Prototype-January-2025?page-id=2143%3A6061&node-id=2146-7246&viewport=-595%2C-659%2C0.22&t=A1CuORABoO0Qexzk-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=2146%3A6646

## Description

This PR includes the following changes:
- Add/Edit form styling updates for admin and user settings tabs
- The vertical tab layout has been updated with icons on the admin page
- The user settings table has been updated to include action buttons on the table directly
- Accordion and other component elevation values have been updated to remove unwanted outlines/shadows

## How Has This Been Tested?

This has been tested locally using docker compose. 

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [ X ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] My changes require new environment variables:
  - [ ] I have updated the docker-compose, K8s YAML, and all dependent deployment configuration files.
- [ ] My changes require updates to the documentation:
  - [ ] I have updated the documentation accordingly.
- [ ] My changes require updates and/or additions to the unit tests:
  - [ ] I have modified/added tests to cover my changes.
- [ ] All existing tests pass.

mwodahl committed:
Update User Settings table to include refresh, add buttons and removed cell borders

mwodahl committed:
Add additional table options to admin RSU table

mwodahl committed:
Remove header from admin page

mwodahl committed:
Update admin tabs to have icons, additional css properties

mwodahl committed:
Remove header, update admin intersection table with additional icons

mwodahl committed:
Update admin user tab with icons, remove header

mwodahl committed:
Remove title, add new background color to admin organization tab

mwodahl committed:
Update admin tab content height

mwodahl committed:
Remove unused css class

mwodahl committed:
Update admin organization page header display

mwodahl committed:
Update organization delete icon

mwodahl committed:
Update admin organization RSU tab table with additional actions, remove title

mwodahl committed:
Add multiselect css overrides

mwodahl committed:
Update intersection actions, remove table title

mwodahl committed:
Remove bottom margin from form-dropdown

mwodahl committed:
Update intersection tab table field width

mwodahl committed:
Update table actions for organization user tab

mwodahl committed:
Remove className from role dropdown

mwodahl committed:
Remove unused css org user class

mwodahl committed:
Update dropdown border override

mwodahl committed:
Update add form styling

mwodahl committed:
Update styling for user, intersection add forms

mwodahl committed:
Update add RSU form styling

mwodahl committed:
Remove unused import

mwodahl committed:
Update add notification tab to use dialog, updated form UI

mwodahl committed:
Update edit notification to use dialog, updated form UI

mwodahl committed:
Remove unused import statements

mwodahl committed:
Add required navigation elements to add notification test

mwodahl committed:
Update testing snapshots

mwodahl committed:
Update add intersection form to use material UI select dropdown

mwodahl committed:
Reorder RSU table row actions

mwodahl committed:
Update add RSU form to use react select components

mwodahl committed:
Update add user form to use react MUI select components

mwodahl committed:
Update org edit modal to use react MUI form components

mwodahl committed:
Update form id

mwodahl committed:
Update incorrect controlId values

mwodahl committed:
Update edit intersection to use react MUI form components

mwodahl committed:
Update edit RSU form to use React MUI form components

mwodahl committed:
Update edit user form to use React MUI form components

mwodahl committed:
Update table header color

mwodahl committed:
Update form input layout, remove unnecessary imports

mwodahl committed:
Update form input layout

mwodahl committed:
Remove unused imports

mwodahl committed:
Remove unused getTitle function

mwodahl committed:
Remove getTitle function

mwodahl committed:
remove getTitle

mwodahl committed:
Update snapshots

mwodahl committed:
Update intersection test to just test for snapshot

mwodahl committed:
Update rsu test to just take a snapshot

mwodahl committed:
Update org delete menu snapshot

mwodahl committed:
Merge branch 'Update/ui-theme' into Update/webapp-admin-user-settings-ui

rgelmis commented:
# Sidebar Updates
## Layout
### Remove custom bottom margin on icon
I think it would be better to use a layout item like a Stack to hold the icon and text here. Then spacing would only have to be set once, rather than hard-coding a value as a part of the switch statement in `getIcon` on line 62 in `VerticalTabs.tsx`. 
### Tab Updates
Remove hard-coded height and min-height, add `padding: 8px 16px`
## Styling
### Borders
Remove side border on active item
### Typography
Change font to Trebuchet, font size to 16px
### Update Icons
RSUs: https://mui.com/material-ui/material-icons/?srsltid=AfmBOooZxl1Kew9WBa3O_f-wcTBuFc7liPb16rtOTXbvewplGRw40NgL&theme=Outlined&query=cell+tower&selected=CellTowerOutlined
Users: https://mui.com/material-ui/material-icons/?srsltid=AfmBOooZxl1Kew9WBa3O_f-wcTBuFc7liPb16rtOTXbvewplGRw40NgL&query=user&theme=Outlined&selected=GroupOutlined
Organizations: https://mui.com/material-ui/material-icons/?srsltid=AfmBOooZxl1Kew9WBa3O_f-wcTBuFc7liPb16rtOTXbvewplGRw40NgL&query=circles&theme=Outlined&selected=WorkspacesOutlined

rgelmis commented:
# Table Updates
These should be global changes for all the relevant tables in the Admin tabs
## Layout
### Padding
All table cells should be set to `padding: 6px 16px`
### Remove "rows" from rows per page dropdown 
![image](https://github.com/user-attachments/assets/71de82eb-7cf3-413b-9238-ac75068005de)
## Styling
### Buttons
- Remove weird hover on all buttons 
![image](https://github.com/user-attachments/assets/3a8c448b-4820-495e-bcfa-7f05ff73b623)
- Action icon buttons should use `#9DBDD3`, updated sizing/padding 
![image](https://github.com/user-attachments/assets/c786bb52-b17d-44d2-89fc-7228bc7100ac)
 and have updated hover 
![image](https://github.com/user-attachments/assets/0a53eefb-0027-4301-9595-a269ca2b0e0f)
### Typography
- Table cells other than column headers should use secondary color and Trebuchet font
- Rows per page/pagination should be Trebuchet 12px and use secondary text color 
- Column headers should not be in all caps, this can be unset using the `.capital-case` class mentioned in the Header UI PR, if you move to a global location

rgelmis commented:
# Form Updates
## Layout
- Add labels to all inputs
- Width: 600px allows for more space between fields
- Remove mb-3 class from fields
- Padding/margin should match between header and footer
- Change "Close" to "Cancel"

## Styling
- Headers should use Trebuchet 16px
- Form fields should use Trebuchet and secondary text color

## Functionality
- Validation is said to be required for some fields but these are not actually validated 
![image](https://github.com/user-attachments/assets/fcca13b3-0ce3-4684-99cd-207e5507da2b)
- Use form props to indicate required fields - we should remove references to bootstrap where we can so there is less confusion about which theme to use/keep styling consistent

rgelmis commented:
# Organizations Tab
## Layout
- Tables should take up the full width of the accordion body 
Current 
![image](https://github.com/user-attachments/assets/c204f518-1ade-4f2a-ba9f-f070d3e1f5ff)
Intended design 
![image](https://github.com/user-attachments/assets/4becde83-fcd9-4664-97ee-4d3a73f0f379)

## Styling
- Remove box shadow from tables 
![image](https://github.com/user-attachments/assets/fe7c8c61-a65a-4821-8491-18dd5565dab0)
- Update styling for the dropdowns, get rid of the weird hover on each 
Current
![image](https://github.com/user-attachments/assets/4e0d841b-4694-4ef4-a1c1-99f70074d9c5)
Intended design
![image](https://github.com/user-attachments/assets/99de5d85-2bc5-4024-8762-f38e9c32ea4b)
- Get rid of weird hover on these buttons 
![image](https://github.com/user-attachments/assets/96e64bed-d828-4a81-a2ea-03274dd69441)


## Functionality
- Not sure what is going on here, might be out of the scope of this PR
![image](https://github.com/user-attachments/assets/a82e8f02-9312-4400-b4cb-a28cdf337884)

rgelmis commented:
This one is ready for ya @mwodahl

mwodahl committed:
Merge branch 'Update/ui-theme' into Update/webapp-admin-user-settings-ui

mwodahl committed:
Update AdminIntersectionTab tableActions to use iconProps

mwodahl committed:
Add missing iconProps to admin pages

payneBrandon commented:
@mwodahl looks like a handful of merge conflicts and comments still open on this one

mwodahl committed:
Add additional classes to App.css

mwodahl committed:
Set color on AdminOrganizationDeleteMenu icon

mwodahl committed:
Update VerticalTabs to use start icon, remove indicator on selected

mwodahl committed:
Remove unnecessary classes from Header.css

mwodahl committed:
Update table actions / icons to display correctly with AdminTable updates

mwodahl committed:
Add rowActionIcon for additional row action icon color

mwodahl committed:
Update style overrides to use text secondary color

mwodahl committed:
Update button style, remove mb-3 class, update header on add/edit notification

mwodahl committed:
Update table actions with iconProps

mwodahl committed:
Update action override to display icon components correctly, add additional styling to table

mwodahl committed:
Update table actions

mwodahl committed:
Update accordionDetails padding

mwodahl committed:
Update table actions to return correct default value

mwodahl committed:
Remove mb-3 class, update input labels, fix padding for Add/Edit RSU

mwodahl committed:
Remove form-dropdown classes

mwodahl committed:
Remove form-dropdown class

mwodahl committed:
Remove mb-3 class, add labels to text fields for Add/Edit intersection

mwodahl committed:
Remove mb-3 class, add labels to text fields for Add/Edit organization

mwodahl committed:
Remove mb-3 class, add labels to text fields for Add/Edit user

mwodahl committed:
Add ThemeProvider to AdminOrganizationDeleteMenu test

mwodahl committed:
Remove museo-slab font-family override

mwodahl committed:
Update outdated testing snapshots

rgelmis reviewed (approved):
lgtm ✨

payneBrandon commented on webapp/src/features/adminEditRsu/AdminEditRsu.tsx:
```diff
@@ -150,262 +159,394 @@ const AdminEditRsu = () => {
     <Dialog open={open}>
-      <DialogTitle>Edit RSU</DialogTitle>
-      <DialogContent>
+      <DialogContent sx={{ width: '600px', padding: '5px 10px' }}>
```
issue: when first clicking "edit" this dialog is hidden and I instead see the "unknown rsu" message popup momentarily before this correct edit page is shown

mwodahl replied:
I talked offline with @drewjj and verified that this was a known issue. I've added a backlog item to address this.

payneBrandon commented on webapp/src/features/adminEditIntersection/AdminEditIntersection.tsx:
```diff
@@ -118,175 +129,252 @@ const AdminEditIntersection = () => {
       {Object.keys(apiData ?? {}).length != 0 ? (
         <>
-          <DialogContent>
+          <DialogContent sx={{ width: '600px', padding: '5px 10px' }}>
```
issue: when editing I momentarily see the "unknown" message box before this correct edit form is shown

mwodahl replied:
Same known issue here - bug added to the backlog

payneBrandon reviewed (changes requested):
A couple issues with the edit pages and momentary unknown display. Otherwise looking good!

mwodahl committed:
Fix delete icon font-size

mwodahl committed:
Fix incorrect reference point longitude error message

jacob6838 commented:
A few comments:
1. all pages "N selected" above the tables need more padding to the left, and default height of the organization page tables is too much, blocks other elements
<img width="1127" alt="image" src="https://github.com/user-attachments/assets/ba10c732-c781-4432-99a5-59def2015d84" />
2. Organizations page new organization and edit page has an extra black box at the top:
<img width="641" alt="image" src="https://github.com/user-attachments/assets/89d210d6-4d3c-47d8-9a80-addac7b41971" />
- On a larger level, i feel like these new/edit pages should just be dialogs over the normal page, no reason to switch pages just to show a dialog with a blank background. But that's a discussion for another time

mwodahl committed:
Update padding on row select, fix display on empty rows

mwodahl committed:
Fix display on add/edit organization

mwodahl committed:
Update outdated Admin/User Settings testing snapshots

mwodahl commented:
> A few comments:
> 
> 1. all pages "N selected" above the tables need more padding to the left, and default height of the organization page tables is too much, blocks other elements
> 
> <img alt="image" width="1127" src="https://private-user-images.githubusercontent.com/51723791/439968811-ba10c732-c781-4432-99a5-59def2015d84.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYyMDMwMDgsIm5iZiI6MTc0NjIwMjcwOCwicGF0aCI6Ii81MTcyMzc5MS80Mzk5Njg4MTEtYmExMGM3MzItYzc4MS00NDMyLTk5YTUtNTlkZWYyMDE1ZDg0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTAyVDE2MTgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxOWIzNWU5YTU1Nzc2OGZhZTZmOWNhMGNjOTQ5Mzg3ZGFhOGMyNTkxYmVmMzM2YWQ3YTJlMjFlMjQxOTQxZWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YmVmmp_R_yG7JyQ1HhjKCOjFKpyLD37gOMN6fatM7d8"> 2. Organizations page new organization and edit page has an extra black box at the top: <img alt="image" width="641" src="https://private-user-images.githubusercontent.com/51723791/439968997-89d210d6-4d3c-47d8-9a80-addac7b41971.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYyMDMwMDgsIm5iZiI6MTc0NjIwMjcwOCwicGF0aCI6Ii81MTcyMzc5MS80Mzk5Njg5OTctODlkMjEwZDYtNGQzYy00N2Q4LTlhODAtYWRkYWM3YjQxOTcxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNTAyVDE2MTgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcxMzAyNDI5ZGFmZTIyMzk4YTg5OGQ0YTc0MTk2MDI3MzMyZGEwYmRlNmNkODQxZjUyNjYyZTFmZWU3YzM1ZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.prPUyuWLQzwC-J5aeEyYhFv51ARZ0dwMifvKGRPAKLY"> - On a larger level, i feel like these new/edit pages should just be dialogs over the normal page, no reason to switch pages just to show a dialog with a blank background. But that's a discussion for another time

I've added padding to the `Rows Selected` text as well as removed empty rows from the table display. I also fixed the issues with the extra black box at the top of the add/edit organization pages.

payneBrandon reviewed (approved):
lgtm!
